server/test/services/cloud/aws/AwsPublicStorageTest.java

```diff
@@ -0,0 +1,164 @@
+package services.cloud.aws;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+import com.typesafe.config.Config;
+import java.io.File;
+import org.junit.Test;
+import play.Environment;
+import play.Mode;
+import repository.ResetPostgres;
+import software.amazon.awssdk.auth.credentials.AwsCredentials;
+import software.amazon.awssdk.auth.credentials.AwsSessionCredentials;
+
+public class AwsPublicStorageTest extends ResetPostgres {
+
+  @Test
+  public void getSignedUploadRequest_prodEnv_actionLinkIsProdAws() {
+    AwsPublicStorage awsPublicStorage =
+        new AwsPublicStorage(
+            instanceOf(AwsStorageUtils.class),
+            instanceOf(AwsRegion.class),
+            instanceOf(Credentials.class),
+            instanceOf(Config.class),
+            new Environment(new File("."), Environment.class.getClassLoader(), Mode.PROD));
+
+    SignedS3UploadRequest uploadRequest =
+        awsPublicStorage.getSignedUploadRequest("fileKey", "redirect");
+
+    assertThat(uploadRequest.actionLink()).contains("amazonaws.com");
+  }
+
+  @Test
+  public void getSignedUploadRequest_devEnv_actionLinkIsLocalStack() {
```

Comment:
question (non-blocking): Should we have a test for the `Test` case as well?

Reply:
Yeah why not 😁  Done.